File: SDKs/amazonaws.com/iot/2015-05-28/python/src/sdk/models/shared/s3action.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import cannedaccesscontrollist_enum as shared_cannedaccesscontrollist_enum


@dataclass_json
@dataclasses.dataclass
class S3Action:
    r"""S3Action
    Describes an action to write data to an Amazon S3 bucket.
    """
    
    bucket_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('bucketName') }})
    key: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('key') }})
    role_arn: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('roleArn') }})
    canned_acl: Optional[shared_cannedaccesscontrollist_enum.CannedAccessControlListEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('cannedAcl') }})
    

File: SDKs/amazonaws.com/iot/2015-05-28/java/lib/src/main/java/openapisdk/models/shared/S3Action.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * S3Action
 * Describes an action to write data to an Amazon S3 bucket.
**/
public class S3Action {
    @JsonProperty("bucketName")
    public String bucketName;
    public S3Action withBucketName(String bucketName) {
        this.bucketName = bucketName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("cannedAcl")
    public CannedAccessControlListEnum cannedAcl;
    public S3Action withCannedAcl(CannedAccessControlListEnum cannedAcl) {
        this.cannedAcl = cannedAcl;
        return this;
    }
    @JsonProperty("key")
    public String key;
    public S3Action withKey(String key) {
        this.key = key;
        return this;
    }
    @JsonProperty("roleArn")
    public String roleArn;
    public S3Action withRoleArn(String roleArn) {
        this.roleArn = roleArn;
        return this;
    }
}